src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/_params.py

```diff
@@ -166,3 +166,11 @@ def get_hostname_completion_list(prefix, action, parsed_args, **kwargs):  # pyli
 # For commands with shared impl between webapp and functionapp and has output, we apply type validation to avoid confusions
 register_cli_argument('appservice web show', 'name', arg_type=webapp_name_arg_type, validator=validate_existing_web_app)
 register_cli_argument('functionapp show', 'name', arg_type=name_arg_type, validator=validate_existing_function_app)
+
+register_cli_argument('functionapp', 'name', arg_type=name_arg_type, id_part='name', help="name of the function")
```

Comment:
If these are just copy/paste (or nearly) of the webapp commands, you can do something like:
```Python
for item in ['functionapp', 'webapp']:
  register_cli_argument('{} create'.format(item), 'plan', ...)
```